File: solutions/typescript/atbash-cipher/1/atbash-cipher.ts
export function encode(plainText: string): string {
    const result: string[] = [];
    let count = 0;

    for (const char of plainText.toLowerCase()) {
        if (char.match(/[0-9]/)) {
            if (count > 0 && count % 5 === 0) {
                result.push(' ');
            }
            result.push(char);
            count++;
            continue;
        }
        if (char.match(/[a-z]/)) {
            if (count > 0 && count % 5 === 0) {
                result.push(' ');
            }
            result.push(String.fromCharCode(122 - (char.charCodeAt(0) - 97)));
            count++;
        }
    }
    return result.join('');
}

export function decode(cipherText: string): string {
    const result: string[] = [];
    for (const char of cipherText) {
        if (char.match(/[0-9]/)) {
            result.push(char);
            continue;
        }
        if (char.match(/[a-z]/)) {
            result.push(String.fromCharCode(122 - (char.charCodeAt(0) - 97)));
        }
    }
    return result.join('');
}

File: solutions/lua/atbash-cipher/1/atbash-cipher.lua
return {
    encode = function(plaintext)
        local result = ""
        local count = 0
        for i = 1, #plaintext do
            local char = plaintext:sub(i, i):lower()
            if char:match("[0-9]") then
                if count > 0 and count % 5 == 0 then
                    result = result .. " "
                end
                result = result .. char
                count = count + 1
                goto continue
            end
            if char:match("[a-z]") then
                if count > 0 and count % 5 == 0 then
                    result = result .. " "
                end
                result = result .. string.char(122 - (char:byte() - 97))
                count = count + 1
            end
            ::continue::
        end
        return result
    end,
}
